ladybug/comfort/pmv.py

```diff
@@ -1,1227 +1,555 @@
+                    * heat loss by sweating
+                    * heat loss by latent respiration
+                    * heat loss by dry respiration
+                    * heat loss by radiation
+                    * heat loss by convection
+
+    References
+    ----------
+    [1] ASHRAE Standard 55 (2010). "Thermal Environmental Conditions
+    for Human Occupancy".
+
+    [2] Hoyt Tyler, Schiavon Stefano, Piccioli Alberto, Cheung Toby, Moon Dustin,
+    and Steinfeld Kyle, 2017, CBE Thermal Comfort Tool. Center for the Built
+    Environment, University of California Berkeley,
+    http://comfort.cbe.berkeley.edu/
+
+    [3] Doherty, T.J., and E.A. Arens.  (1988).  Evaluation of the physiological
+    bases of thermal comfort models. ASHRAE Transactions, Vol. 94, Part 1, 15 pp.
+    https://escholarship.org/uc/item/6pq3r5pr
     """
 
-    def __init__(self, air_temperature=None, rad_temperature=[], wind_speed=[],
-                 rel_humidity=[], met_rate=[], clo_value=[], external_work=[]):
-        """Initialize a PMV comfort object from lists of PMV inputs."""
-        # Assign all of the input values to the PMV comfort model object.
-        # And assign defaults if nothing has been connected.
-        self.air_temperature = air_temperature
+    result = {}
+    set = pierce_set(ta, tr, vel, rh, met, clo, wme)
+
+    if vel <= still_air_threshold:
+        pmv, ppd, heat_loss = pmv_fanger(ta, tr, vel, rh, met, clo, wme)
+        ta_adj = ta
+        ce = 0
+    else:
+        ce_l = 0
+        ce_r = 40
+        eps = 0.001  # precision of ce
+
+        def fn(ce):
+            return (set - pierce_set(
+                ta - ce, tr - ce,
+                still_air_threshold, rh, met, clo, wme))
+
+        ce = secant(ce_l, ce_r, fn, eps)
+        if ce == 'NaN':
+            ce = bisect(ce_l, ce_r, fn, eps, 0)
+
+        pmv, ppd, heat_loss = pmv_fanger(
+            ta - ce, tr - ce, still_air_threshold, rh, met, clo, wme)
+        ta_adj = ta - ce
+
+    result['pmv'] = pmv
+    result['ppd'] = ppd
+    result['set'] = set
+    result['ta_adj'] = ta_adj
+    result['ce'] = ce
+    result['heat_loss'] = heat_loss
+
+    return result
+
+
+def pmv_fanger(ta, tr, vel, rh, met, clo, wme=0):
+    """Calculate PMV using only Fanger's original equation.
+
+    Note that Fanger's original expereiments were only conducted at
+    low air speeds (<0.1 m/s) and the pmv() function above should be
+    utilized in the case that air speeds may be higher than 0.1 m/s.
+
+    Args:
+        ta: air temperature [C]
+        tr: mean radiant temperature [C]
+        vel: relative air velocity [m/s]
+        rh: relative humidity [%]
+        met: metabolic rate [met]
+        clo: clothing [clo]
+        wme: external work [met], normally around 0 when seated
+
+    Returns:
+        pmv: predicted mean vote (PMV)
+        ppd: percentage of people dissatisfied (PPD) [%]
+        heat_loss: A list with the 6 heat loss terms of the PMV model [W].
```

Comment:
I would seriously consider taking this out, then, which would be a shame as I know several people are interested in this information.  Still, the outputs from this method really need to be small since this method will be run literally hundreds of thousands of times for an annual comfort map.  Having a dictionary with keys is going to significantly increase the memory requirements here. 

Or maybe, if I can't use a tuple or a list, I'll add another optional input to the method that allows you to request whether you want the heat loss terms or not?  Then I can output them as a dictionary.  Would that be better?

Reply:
That's a fair point but doesn't justify removing it. In that case I think we should implement another version of the method which just returns minimum outputs that you will need for an annual comfort map and make the outputs for this one easier to access.